Gems/Atom/Feature/Common/Code/Source/SkyAtmosphere/SkyAtmospherePass.cpp

```diff
@@ -200,8 +201,40 @@ namespace AZ::Render
         }
     }
 
+    bool SkyAtmospherePass::NeedsShaderDataRebuild() const
+    {
+        if (m_children.empty())
+        {
+            return false;
+        }
+        else if (m_children.size() != m_atmospherePassData.size())
+        {
+            return true;
+        }
+
+        // SRG may change due to a shader reload
+        for (int i = 0; i < m_atmospherePassData.size(); ++i)
+        {
+            if (RPI::RenderPass* renderPass = azrtti_cast<RPI::RenderPass*>(m_children[i].get()))
+            {
+                auto srg = renderPass->GetShaderResourceGroup();
+                if (m_atmospherePassData[i].m_srg != srg)
+                {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
     void SkyAtmospherePass::FrameBeginInternal(AZ::RPI::Pass::FramePrepareParams params)
     {
+        if (NeedsShaderDataRebuild())
+        {
+            BuildShaderData();
```

Comment:
I think you need to set m_updateConstants to true after the shader data was rebuilt

Reply:
oh, I moved that line inside of `BuildShaderData` since I always need to update the constants after building the shader data.